File: 백준/Silver/10828. 스택/스택.py
import sys


input = lambda: sys.stdin.readline().rstrip()

N = int(input())

stack = []
result = []
for _ in range(N):
    data = input().split()
    if data[0] == "push":
        x = int(data[1])
        stack.append(x)
    elif data[0] == "pop":
        if stack:
            result.append(stack.pop())
        else:
            result.append(-1)
    elif data[0] == "size":
        result.append(len(stack))
    elif data[0] == "empty":
        result.append(int(not stack))
    else:
        if stack:
            result.append(stack[-1])
        else:
            result.append(-1)

print("\n".join(map(str, result)))

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int N;
    cin >> N;
    stack<int> s;

    while (N--)
    {
        string command;
        cin >> command;
        if (command == "push")
        {
            int X;
            cin >> X;
            s.push(X);
        }
        else if (command == "pop")
        {
            if (s.empty())
            {
                cout << "-1\n";
            }
            else
            {
                cout << s.top() << '\n';
                s.pop();
            }
        }
        else if (command == "size")
        {
            cout << s.size() << '\n';
        }
        else if (command == "empty")
        {
            if (s.empty())
            {
                cout << "1\n";
            }
            else
            {
                cout << "0\n";
            }
        }
        else if (command == "top")
        {
            if (s.empty())
            {
                cout << "-1\n";
            }
            else
            {
                cout << s.top() << '\n';
            }
        }
    }
}